tutorials/src/bin/basic-tutorial-4.rs

```diff
@@ -0,0 +1,138 @@
+extern crate gstreamer as gst;
+use std::io;
+use std::io::Write;
+use gst::prelude::*;
+use gst::MessageView;
+
+struct CustomData {
+    playbin: gst::Element, // Our one and only element
+    playing: bool, // Are we in the PLAYING state?
+    terminate: bool, // Should we terminate execution?
+    seek_enabled: bool, // Is seeking enabled for this media?
+    seek_done: bool, // Have we performed the seek already?
```

Comment:
These bool flags actually encode a state machine, right? So ideally that could be an enum for the current state. But let's keep it like it is in C